src/App.jsx

```diff
@@ -14,33 +16,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_LENGTH = 'sortByLength';
+const SORT_ALPHABETICALLY = 'sortAlphabetically';
+const REVERSE_LIST = 'reverseList';
+const RESET = 'reset';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoodsFromServer(goods, buttonId) {
+  switch (buttonId) {
+    case SORT_ALPHABETICALLY:
+      return [...goods].sort((good1, good2) => good1.localeCompare(good2));
+    case SORT_BY_LENGTH:
+      return [...goods].sort((good1, good2) => good1.length - good2.length);
+    case REVERSE_LIST:
+      return [...goodsFromServer].reverse();
+    case RESET:
+      return [...goodsFromServer];
+    default:
+      return goods;
+  }
+}
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+export const App = () => {
+  const [visibleVegetables, setVisibleVegetables] = useState([
+    ...goodsFromServer,
+  ]);
+  const [buttonClassSortAlpha, setButtonClassSortAlpha] =
+    useState('is-info is-light');
+  const [buttonClassSortByLength, setButtonClassSortByLength] = useState(
+    'is-success is-light',
+  );
+  const [buttonClassReverse, setButtonClassReverse] = useState(
+    'is-warning is-light',
+  );
+  const [buttonClassReset, setButtonClassReset] =
+    useState('is-danger is-light');
```

Comment:
Looks very strange, just create state for sortField and store sort value here. Later you can reuse it with classnames library to change color of button.